src/main/java/com/aws/greengrass/lifecyclemanager/UpdateSystemPolicyService.java

```diff
@@ -47,7 +48,7 @@
     // String identifies the action, the pair consist of timeout and an action. The timeout
     // represents the value in seconds the kernel will wait for components to respond to
     // an precomponent update event
-    private final Map<String, UpdateAction> pendingActions = new LinkedHashMap<>();
+    private final Map<String, UpdateAction> pendingActions = Collections.synchronizedMap(new LinkedHashMap<>());
```

Comment:
Why not just use ConcurrentHashMap?

Reply:
We need to keep ordering I believe

Reply:
It needs to be a LinkedHashMap so all pending actions are executed in order, right now we only ever have 1 action because we use this only for deployments and we process 1 deployment at a time, but this service was written to accommodate more actions and execute them in order of registration, and we've always left it like that